apm-agent-plugins/apm-vertx/apm-vertx3-plugin/src/main/java/co/elastic/apm/agent/vertx/v3/web/http1/HttpServerRequestImplStartInstrumentation.java

```diff
@@ -70,10 +77,14 @@ public static Object enter(@Advice.This HttpServerRequestImpl request) {
         public static void exit(@Advice.This HttpServerRequestImpl request,
                                 @Advice.Enter Object transactionObj,
                                 @Advice.Thrown @Nullable Throwable thrown) {
-            if (transactionObj instanceof Transaction) {
-                Transaction transaction = (Transaction) transactionObj;
-                transaction.captureException(thrown).deactivate();
+            if (!(transactionObj instanceof Transaction)) {
+                return;
             }
+            Transaction transaction = (Transaction) transactionObj;
+            transaction.captureException(thrown).deactivate();
+
+            HttpServerResponseImpl response = request.response();
+            response.endHandler(new ResponseEndHandlerWrapper(transaction, response, request));
```

Comment:
this replaces the constructor instrumentation as we can access the response directly once the original method body has been executed.